app/server/apiGatewayHandler.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  StackResourceSummaries,
+  StackResourceSummary
+} from "aws-sdk/clients/cloudformation";
+import express from "express";
+import fetch from "node-fetch";
+import url from "url";
+import { MDA_TEST_USER_HEADER } from "../shared/productResponse";
+import { APIGateway, AWS_REGION, CloudFormation } from "./awsIntegration";
+import { conf } from "./config";
+import { log } from "./log";
+
+const isProd = conf.STAGE.toUpperCase() === "PROD";
+
+const normalUserApiStage = isProd ? "PROD" : "DEV"; // i.e. CODE manage-frontend uses DEV when in normal mode
+const testUserApiStage = "CODE";
+
+const apiNames = [
+  "cancellation-sf-cases-api",
+  "delivery-records-api",
+  "holiday-stop-api"
+] as const;
+export type ApiName = typeof apiNames[number];
+
+const byResourceType = (resourceTypeFilter: string) => (
+  resource: StackResourceSummary
+) => resource.ResourceType === resourceTypeFilter;
+
+const toDefinedPhysicalResourceId = (stackName: string) => (
+  resource: StackResourceSummary
+) => {
+  if (resource.PhysicalResourceId) {
+    return resource.PhysicalResourceId;
+  }
+  throw new Error(
+    `PhysicalResourceId missing for '${resource.ResourceType}' of ${stackName}`
+  );
+};
+
+const getHost = (
+  stackName: string,
+  stage: string,
+  stackResourceSummaries?: StackResourceSummaries
+) => {
+  const hosts = stackResourceSummaries
+    ?.filter(byResourceType("AWS::ApiGateway::RestApi"))
+    .map(toDefinedPhysicalResourceId(stackName))
+    .map(
+      apiGatewayId =>
+        `${apiGatewayId}.execute-api.${AWS_REGION}.amazonaws.com/${stage}`
+    );
+  if (hosts && hosts.length === 1) {
+    return hosts[0];
+  }
+  log.error(`${(hosts || []).length} hosts for ${stackName}, expected 1`);
+};
+
+const lookupApiKey = async (apiKey: string) =>
+  (
+    await APIGateway.getApiKey({
+      apiKey,
+      includeValue: true
+    }).promise()
+  ).value;
+
+const getApiKeyPromise = (
+  stackName: string,
+  stage: string,
+  stackResourceSummaries?: StackResourceSummaries
+) => {
+  const apiKeyPromises = stackResourceSummaries
+    ?.filter(byResourceType("AWS::ApiGateway::ApiKey"))
+    .map(toDefinedPhysicalResourceId(stackName))
+    .map(lookupApiKey);
+  if (apiKeyPromises && apiKeyPromises.length === 1) {
+    return apiKeyPromises[0];
+  }
+  log.error(
```

Comment:
Why is `log.error` sufficient instead of  `throw`?

Reply:
this would just log the error, at launch (something like aws-sdk permission issues or no stack with that name existing).
We don't want to block the start-up of the entire of manage just because we can't get the host/api key for a service.